src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final Circle defaultCircle = new Circle(10, Color.WHITE);
+
+    public Figure getRandomFigure() {
+        int randomFigureType = random.nextInt(5);
+        int randomSide = random.nextInt() * 16;
+        int randomMultiplier = random.nextInt(3) + 2;
+
+        switch (randomFigureType) {
+            case 0:
+                return new Square(randomSide, randomColor.getRandomColor());
+            case 1:
+                return new Rectangle(randomSide,
+                        randomSide * randomMultiplier, randomColor.getRandomColor());
+            case 2:
+                return new RightTriangle(randomSide,
+                        randomSide * randomMultiplier, randomColor.getRandomColor());
+            case 3:
+                return new Circle(randomSide, randomColor.getRandomColor());
+            case 4:
+                return new IsoscelesTrapezoid(randomSide, randomSide,
+                        randomSide / 1.1, randomColor.getRandomColor());
+            default:

```

Comment:
Each side should be randomly generated, your sides are still linked somehow and depend on each other. It should always be a newly generated independent number. You'll need to create more variables to resolve the issue